pypicloud/cache/dynamo.py

```diff
@@ -22,6 +25,78 @@
 LOG = logging.getLogger(__name__)
 
 
+def _path_depth(package):
+    """Count the slashes in path, or return infinity on no path."""
+    path = package.data.get('path')
+    if not path:
+        return float('+inf')
+    return path.count('/')
+
+
+def _decide_between_versions(contender, current):
+    """Decide between packages with the same filename on different paths."""
+
+    if contender.last_modified < current.last_modified:
+        return contender
```

Comment:
Sorry for the belated answer.

I think we're actually defending the same thing. Let me write some tests to show that.

Meanwhile, @patricklucas also found a bug when rebuilding packages with the same package name and version, but different filenames (like wheels). Moving to fix that soon.